addons/website_sale/controllers/main.py

```diff
@@ -272,8 +272,8 @@ def product(self, product, category='', search='', **kwargs):
 
         if not context.get('pricelist'):
             context['pricelist'] = int(self.get_pricelist())
-            product = template_obj.browse(cr, uid, int(product), context=context)
-
+            # Public user should able to read virtual stock
+            product = template_obj.browse(cr, SUPERUSER_ID, int(product), context=context)
```

Comment:
Is that really necessary ? Why should you read as SUPERUSER_ID for virtual stock ?